site/tests/app/controllers/admin/PlagiarismControllerTester.php

```diff
@@ -0,0 +1,34 @@
+<?php namespace tests\app\controllers\admin;
+
+use app\libraries\Core;
+use tests\BaseUnitTest;
+use app\controllers\admin\PlagiarismController;
+
+class PlagiarismControllerTester extends BaseUnitTest
+{
+   /** @var PlagiarismController */
+    protected $controller;
+
+    /** @var Core */
+    protected $core;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->core = $this->createMockCore([], [], [
```

Comment:
You also need to pass some config when creating the core mock to set the semester and course to the file you create (and delete after probably). This is because the controller has this code
`$semester = $this->core->getConfig()->getSemester();
$course = $this->core->getConfig()->getCourse();`